### PR TITLE
security: fix removing trust anchors

### PR DESCRIPTION
**Pull request by Danielius1922**
Add test to verify consistency of internal lists containing
identity certificates.

### CODE REVIEW

File: security/oc_pki.c [hunk 1]
```diff
@@ -119,7 +119,7 @@ pki_add_intermediate_cert(size_t device, int credid, const unsigned char *cert,
 
   if (ret > 0) {
     OC_DBG("added intermediate CA cert to /oic/sec/cred");
-    oc_tls_refresh_identity_certs();
+    oc_tls_add_new_identity_certs();
```
Comment (jkralik) [inline security/oc_pki.c:122]: `oc_tls_add_new_identity_cert` this is weird name ... I expected cred as arg

File: security/oc_tls.h [hunk 2]
```diff
@@ -155,16 +155,78 @@ void oc_tls_use_pin_obt_psk_identity(void);
...
-void oc_tls_remove_identity_cert(oc_sec_cred_t *cred);
+/**
+ * @brief Internal interface for adding new identity certificate chains.
+ *
+ * Iterate over all credentials, check if a credential is associated with a leaf
+ * identity certificate. If the identity certificate doesn't exist in the global
+ * list of identity certificates then create a new identity certificate item and
+ * add it to the list.
+ */
+void oc_tls_add_new_identity_certs(void);
```
Comment (jkralik) [inline security/oc_tls.h:166]: `oc_tls_add_new_identity_certs` -> `oc_tls_resolve_identity_certs` ?

**Review by jkralik — APPROVED** [review]

File: security/oc_tls.h (hunk 2 above)
Comment (Danielius1922) [reply security/oc_tls.h]: sure that works too

**Review by jkralik — APPROVED** [review]

**Comment by ondrejtomcik** [comment]
@WAvdBeek order n1

**Review by WAvdBeek — APPROVED** [review]

**Pull request merged**